sw/device/lib/dif/README.md

```diff
@@ -50,6 +51,30 @@ there are some relaxations of these rules for them described at the end.
 * DIFs must use enum return codes rather than booleans for reporting errors. If
   a DIF can either error or instead produce a value, it should return an enum
   return code, and use an out-parameter for returning the produced value.
+  * DIF libraries must have a general return code enum called
+    `dif_<ip>_result_t`. It must define the following constants (in this order):
+    * `kDif<ip>Ok`, with value 0, to denote the call succeeded.
+    * `kDif<ip>Error`, to denote a non-specific error happened during the call.
+      This is for the `default:` case of enum switches (as noted below), and for
+      assertion errors (usually where the function has already caused side
+      effects so `kDif<ip>BadArg` cannot be used).
+    * `kDif<ip>BadArg`, to denote that the caller supplied incorrect arguments.
```

Comment:
Given what we've done in the UART DIF for configuration, my feeling is if there's a more specific reason an argument is bad (in the UART example, you may have chosen values that set the NCO to an unrepresentable value), then you can return a more specific result constant.

However, I think it is good to always have `BadArg` in the enums, so that things like passing a NULL `dif_<ip>_t*` (which usually includes the base address of the device) can still use that return code (given it will in the generic functions).